test/integration/kongingress_webhook_test.go

```diff
@@ -0,0 +1,88 @@
+//go:build integration_tests
+// +build integration_tests
+
+package integration
+
+import (
+	"testing"
+
+	"github.com/kong/kubernetes-testing-framework/pkg/clusters/types/kind"
+	"github.com/samber/lo"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	admregv1 "k8s.io/api/admissionregistration/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/net"
+
+	configurationv1 "github.com/kong/kubernetes-ingress-controller/v2/pkg/apis/configuration/v1"
+	"github.com/kong/kubernetes-ingress-controller/v2/pkg/clientset"
+	"github.com/kong/kubernetes-ingress-controller/v2/pkg/clientset/scheme"
+)
+
+func TestKongIngressValidationWebhook(t *testing.T) {
+	ns, cleaner := setup(t)
+	defer func() { assert.NoError(t, cleaner.Cleanup(ctx)) }()
+
+	if env.Cluster().Type() != kind.KindClusterType {
+		t.Skip("webhook tests are only available on KIND clusters currently")
+	}
+
+	closer, err := ensureAdmissionRegistration(
+		"kong-validations-kongingress",
+		[]admregv1.RuleWithOperations{
+			{
+				Rule: admregv1.Rule{
+					APIGroups:   []string{"configuration.konghq.com"},
+					APIVersions: []string{"v1"},
+					Resources:   []string{"kongingresses"},
+				},
+				Operations: []admregv1.OperationType{admregv1.Create, admregv1.Update},
+			},
+		},
+	)
+	assert.NoError(t, err, "creating webhook config")
+	defer func() {
+		assert.NoError(t, closer())
+	}()
+
+	t.Log("waiting for webhook service to be connective")
+	err = waitForWebhookServiceConnective(ctx, "kong-validations-kongingress")
+	require.NoError(t, err)
+
+	kongClient, err := clientset.NewForConfig(env.Cluster().Config())
+	require.NoError(t, err)
+
+	t.Run("when deprecated fields are populated warnings are returned", func(t *testing.T) {
+		kongIngress := &configurationv1.KongIngress{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "kong-ingress-validation-",
+			},
+			// Proxy field is deprecated, expecting warning for it.
+			Proxy: &configurationv1.KongIngressService{
+				Protocol: lo.ToPtr("tcp"),
```

Comment:
`address.Of` :) ?

https://github.com/Kong/kubernetes-ingress-controller/blob/ba97e9ee46d912eca610e8da23bb2ad89fc82100/internal/util/address/of.go#L3-L6

```suggestion
				Protocol: address.Of("tcp"),
```

Reply:
a51abda, although we could think about dropping `address.Of` - project already imports `github.com/samber/lo` which implements the same function. WDYT?

Reply:
😬 I put sooooo much effort into this :)

I'm fine with this but we could make this properly e.g. blocking usages of `k8s.io/utils/pointer` and suggesting to use `lo.ToPtr()` instead. Might be done in a separate PR 👍

Reply:
> I'm fine with this but we could make this properly e.g. blocking usages of k8s.io/utils/pointer and suggesting to use lo.ToPtr() instead. Might be done in a separate PR 👍

Yeah, totally agree.